src/libostree/ostree-repo-pull.c

```diff
@@ -151,6 +153,8 @@ typedef struct {
    * whether to fetch the primary object after fetching its
    * detached metadata (no need if it's already stored). */
   gboolean     object_is_stored;
+
+  OstreeCollectionRef *requested_ref;
```

Comment:
Yeah. It can be NULL when pulling just a checksum or when pulling a non-commit object.